gamestate.js

```diff
@@ -123,28 +123,23 @@ function updateValue(elt, newValue, toggleOnZero) {
 }
 
 // Update life totals, with debouncing
-var updateP1Numbers = _.debounce(function(v) {
-    updateValue($(".p1.life"), v.val().life);
-    updateValue($(".p1.poison"), v.val().poison, true);
+function updateSide(prefix, v) {
+    updateValue($(prefix + ".life"), v.val().life);
+    updateValue($(prefix + ".poison"), v.val().poison, true);
     if (v.val().gamewins > 0) {
-        updateValue($(".p1.wins"), v.val().gamewins);
+        updateValue($(prefix + ".wins"), v.val().gamewins);
     } else {
-        $(".p1.wins").text(v.val().gamewins);
+        $(prefix + ".wins").text(v.val().gamewins);
     }
-}, 1000);
+}
 
-var updateP2Numbers = _.debounce(function(v) {
-    updateValue($(".p2.life"), v.val().life);
-    updateValue($(".p2.poison"), v.val().poison, true);
-    if (v.val().gamewins > 0) {
-        updateValue($(".p2.wins"), v.val().gamewins);
-    } else {
-        $(".p2.wins").text(v.val().gamewins);
-    }
+var updateAllNumbers = _.debounce(function () {
+    firebase.database().ref('player1').once('value', updateSide.bind(null, ".p1"));
```

Comment:
Oh nice, I was only looking at the docs for once() which don't mention that.  That's super handy.